google/cloud/internal/populate_common_options.cc

```diff
@@ -57,6 +52,11 @@ Options PopulateCommonOptions(Options opts, std::string const& endpoint_env_var,
       opts.set<EndpointOption>(*std::move(e));
     }
   }
+  if (!opts.has<EndpointOption>()) {
+    absl::StrAppend(&default_endpoint, ".");
```

Comment:
This assumes `default_endpoint` never has a trailing `.`, the previous version did not assume that.

Reply:
We are in control of the callers. We never include the trailing `.`. If we did include the trailing `.`, the `AuthorityOption` would break.